solrmonitor/zkdispatcher.go

```diff
@@ -120,19 +117,33 @@ func (d *ZkDispatcher) eventLoop() {
 	defer atomic.AddInt32(&d.runningProcs, -1)
 
 	for {
-		chosen, recv, recvOK := reflect.Select(d.selectCases)
-		if chosen == 0 {
-			if recvOK {
-				// New handler
-				nh := recv.Interface().(newHandler)
+		// first try to drain any new handlers, bailing if dispatcher is closed
+		done := false
+		for !done {
```

Comment:
sorry this is a total nit, but you can move done into a loop var...

`for done := false; !done; {`

apologies for my OCD